### PR TITLE
[CatalogBackend] Add API to get location by entity

### PR DESCRIPTION
**Pull request by alde**
Introduce API route to get location by entity. Solves #21826.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend/src/modules/core/DefaultLocationStore.ts
Comment (Rugvip) [inline plugins/catalog-backend/src/modules/core/DefaultLocationStore.ts:147]: Might be worth using `parseLocationRef` to get a proper error if this ends up being a location ref of a form that we don't know how to parse. Not that that really is supposed to be able to happen 😅. In particular if the location doesn't contain a `:` the rest of this might get a bit strange.

File: packages/catalog-client/api-report.md
Comment (Rugvip) [inline packages/catalog-client/api-report.md:54]: Perhaps `string | CompoundEntityRef` here, just like `getEntityByRef`?

**Review by Rugvip — APPROVED** [review]
Nice! 😁 🎉 

Just a few thoughts

File: plugins/catalog-backend/src/modules/core/DefaultLocationStore.ts
Comment (alde) [reply plugins/catalog-backend/src/modules/core/DefaultLocationStore.ts:147]: thanks, was unfamiliar with that helper

**Comment by Rugvip** [comment]
Wanna note that it's up to @backstage/catalog-maintainers to merge this as this point

File: .changeset/large-moons-speak.md
Comment (colinodell) [inline .changeset/large-moons-speak.md:2]: This PR [added a new required method to the exported `CatalogApi` interface](https://github.com/backstage/backstage/pull/21975/files#diff-47ce190784b257e830ecee68d94a53cda265056243d13ecfda310cc43723fe9c).  Anyone already implementing that interface in their code would therefore need to add that method to remain compliant with that interface.  Wouldn't that be a breaking change, thus requiring a `major` version bump per semver?

**Pull request merged**
